motionplan/motionPlanner.go

```diff
@@ -19,24 +20,112 @@ import (
 	vutil "go.viam.com/rdk/utils"
 )
 
-// MotionPlanner provides an interface to path planning methods, providing ways to request a path to be planned, and
+// motionPlanner provides an interface to path planning methods, providing ways to request a path to be planned, and
 // management of the constraints used to plan paths.
-type MotionPlanner interface {
+type motionPlanner interface {
 	// Plan will take a context, a goal position, and an input start state and return a series of state waypoints which
 	// should be visited in order to arrive at the goal while satisfying all constraints
-	Plan(context.Context, spatialmath.Pose, []frame.Input, *PlannerOptions) ([][]frame.Input, error)
+	Plan(context.Context, spatialmath.Pose, []frame.Input, *plannerOptions) ([][]frame.Input, error)
 	Frame() frame.Frame // Frame will return the frame used for planning
 }
 
-type plannerConstructor func(frame.Frame, int, golog.Logger) (MotionPlanner, error)
+type (
+	plannerConstructor       func(frame.Frame, int, golog.Logger) (motionPlanner, error)
```

Comment:
[nit] probably no point including this anymore if we're making the single point of entry you are proposing

Reply:
Perhaps we move them into a test file though. They're used for unit tests.

Reply:
Third option: seededPlannerConstructor is actually used in the real code, to make the planner for a waypoint. So we should leave that. While plannerConstructor isn't used at all really; I've removed it from tests and the code entirely.